### PR TITLE
TINY-8209: Remove imagetools plugin

### PR DESCRIPTION
**Pull request by JamesToohey**
Related Ticket: TINY-8209

Description of Changes:
- Remove `imagetools` plugin from TinyMCE monorepo

Pre-checks:
* [x] Changelog entry added
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/plugins/help/main/ts/data/PluginUrls.ts [hunk 1]
```diff
@@ -72,6 +71,7 @@ const urls = Arr.map<PartialPluginUrl, PluginUrl>([
   { key: 'casechange', name: 'Case Change', type: PluginType.Premium },
   { key: 'checklist', name: 'Checklist', type: PluginType.Premium },
   { key: 'export', name: 'Export', type: PluginType.Premium },
+  { key: 'imagetools', name: 'Image Tools', type: PluginType.Premium },
```
Comment (JamesToohey) [inline modules/tinymce/src/plugins/help/main/ts/data/PluginUrls.ts:74]: Just checking this is appropriate

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/contextmenu/MobileContextMenuTest.ts [hunk 2]
```diff
@@ -163,31 +159,6 @@ describe('browser.tinymce.themes.silver.editor.contextmenu.MobileContextMenuTest
     await pWaitForAndCloseDialog(editor);
   });
 
-  it('TBA: Test context menus on image inside a table', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/contextmenu/MobileContextMenuTest.ts:166]: This shouldn't have been removed as there was still some parts here that aren't imagetools related.

- [ ] Revert this and only remove the imagetools specific logic (e.g `selectors.editimage`)

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/contextmenu/DesktopContextMenuTest.ts [hunk 3]
```diff
@@ -120,36 +112,6 @@ describe('browser.tinymce.themes.silver.editor.contextmenu.DesktopContextMenuTes
     await pWaitForAndCloseDialog(editor);
   });
 
-  it('TBA: Test context menus on image inside a table', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/contextmenu/DesktopContextMenuTest.ts:123]: Likewise this shouldn't have been removed.

- [ ] Revert and only remove imagetools specific logic

File: modules/tinymce/src/plugins/help/main/ts/data/PluginUrls.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/help/main/ts/data/PluginUrls.ts:74]: LGTM :+1:

**Review by lnewson — APPROVED** [review]
Looks good, just a few things that shouldn't have been removed. Also don't forget to add a `Removed` changelog entry for this (I know it's not checked, so just clarifying where I think it should go).

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/contextmenu/DesktopContextMenuTest.ts [hunk 4]
```diff
@@ -170,23 +160,4 @@ describe('browser.tinymce.themes.silver.editor.contextmenu.DesktopContextMenuTes
     pressDownArrowKey();
     await pAssertFocusOnItem('Delete Table', '.tox-collection__item:contains("Delete table")');
   });
-
-  it('TBA: Test context menus on placeholder image inside a table', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/contextmenu/DesktopContextMenuTest.ts:174]: Sorry if my previous comment wasn't clear, but this also shouldn't have been removed.

File: modules/tinymce/src/themes/silver/main/ts/ui/menus/contextmenu/Settings.ts [hunk 5]
```diff
@@ -29,7 +29,7 @@ const getMenuItems = (editor: Editor, name: string, defaultItems: string): strin
 const isContextMenuDisabled = (editor: Editor): boolean => editor.getParam('contextmenu') === false;
 
 const getContextMenu = (editor: Editor): string[] => {
-  return getMenuItems(editor, 'contextmenu', 'link linkchecker image imagetools table spellchecker configurepermanentpen');
+  return getMenuItems(editor, 'contextmenu', 'link linkchecker image table spellchecker configurepermanentpen');
```
Comment (ltrouton) [inline modules/tinymce/src/themes/silver/main/ts/ui/menus/contextmenu/Settings.ts:32]: Just wanted to check if this should have been removed? Could be misunderstanding the code, but we still want imagetools to show in the default context menu even when it becomes a premium plugin?

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/themes/silver/main/ts/ui/menus/contextmenu/Settings.ts (hunk 5 above)
Comment (TheSpyder) [reply modules/tinymce/src/themes/silver/main/ts/ui/menus/contextmenu/Settings.ts:32]: correct

- [ ] revert change

**Review by TheSpyder — APPROVED** [review]

**Pull request merged**